front/pages/api/w/[wId]/data_sources/[dsId]/tables/csv.ts

```diff
@@ -91,181 +70,81 @@ async function handler(
-        api_error: {
-          type: "invalid_rows_request_error",
-          message: `Failed to parse CSV: ${csvRowsRes.error.message}`,
-        },
-        status_code: 400,
-      });
-    }
-
-    const enqueueRes = await enqueueUpsertTable({
-      upsertTable: {
-        workspaceId: owner.sId,
-        dataSourceId: dataSource.sId,
-        tableId,
-        tableName: name,
-        tableDescription: description,
-        tableTimestamp: bodyValidation.right.timestamp ?? null,
-        tableTags: bodyValidation.right.tags ?? [],
-        tableParents,
-        csv: csv ?? null,
-        truncate,
-        useAppForHeaderDetection,
-      },
-    });
-    if (enqueueRes.isErr()) {
-      return apiError(
-        req,
-        res,
-        {
-          status_code: 500,
-          api_error: {
-            type: "data_source_error",
-            message:
-              "There was an error enqueueing the the document for asynchronous upsert.",
-          },
-        },
-        enqueueRes.error
-      );
-    }
-    return res.status(200).json({
-      table: {
-        table_id: tableId,
-      },
-    });
-  }
-
-  const tableRes = await upsertTableFromCsv({
-    auth,
-    dataSource,
-    tableId,
-    tableName: name,
-    tableDescription: description,
-    tableTimestamp: bodyValidation.right.timestamp ?? null,
-    tableTags: bodyValidation.right.tags || [],
-    tableParents,
-    csv: csv ?? null,
-    truncate,
-    useAppForHeaderDetection,
-  });
-
-  if (tableRes.isErr()) {
-    if (tableRes.error.type === "internal_server_error") {
-      return apiError(req, res, {
-        api_error: {
-          type: "internal_server_error",
-          message: tableRes.error.message,
-        },
-        status_code: 500,
-      });
-    }
-
-    if (tableRes.error.type === "invalid_request_error") {
-      if ("csvParsingError" in tableRes.error) {
+    case "POST": {
+      const bodyValidation = UpsertTableFromCsvRequestSchema.decode(req.body);
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
         return apiError(req, res, {
           api_error: {
             type: "invalid_request_error",
-            message: `Failed to parse CSV: ${tableRes.error.csvParsingError.message}`,
-          },
-          status_code: 400,
-        });
-      } else if ("inputValidationError" in tableRes.error) {
-        return apiError(req, res, {
-          api_error: {
-            type: "invalid_request_error",
-            message: `Invalid request body: ${tableRes.error.inputValidationError}`,
+            message: `Invalid request body: ${pathError}`,
           },
           status_code: 400,
         });
+      }
+
+      const params = bodyValidation.right;
+
+      const r = await handleDataSourceTableCSVUpsert({
+        auth,
+        params,
+        dataSource,
+      });
+      if (r.isErr()) {
+        if (r.error.code === "missing_csv") {
```

Comment:
`nit` switch/case? To ensure we update this part if we change the logic of `handleDataSourceTableCSVUpsert`